script/src/verify.rs

```diff
@@ -1229,6 +1380,152 @@ fn run_vms(
     }
 }
 
+// Run a series of VMs with control signal, will only return when verification finished
+// Or send `Stop` command when verification is suspended
+async fn run_vms_with_signal(
+    script_group: &ScriptGroup,
+    machines: Vec<ResumableMachine>,
+    context: Arc<Mutex<MachineContext>>,
+    signal: &mut Receiver<ChunkCommand>,
+) -> Result<ChunkState, ScriptError> {
+    if machines.is_empty() {
+        return Err(ScriptError::Other(
+            "To resume VMs, at least one VM must be available!".to_string(),
+        ));
+    }
+
+    let mut pause = machines[0].pause();
+    let (finished_send, mut finished_recv) =
+        mpsc::unbounded_channel::<(Result<i8, ckb_vm::Error>, u64)>();
+
+    // send initial `Resume` command to child
+    // it's maybe useful to set initial command to `signal.borrow().to_owned()`
+    // so that we can control the initial state of child, which is useful for testing purpose
+    let (child_sender, child_recv) = watch::channel(ChunkCommand::Resume);
+    let jh =
+        tokio::spawn(
+            async move { run_vms_child(machines, child_recv, finished_send, context).await },
+        );
+
+    loop {
+        tokio::select! {
+            _ = signal.changed() => {
+                let command = signal.borrow().to_owned();
+                match command {
+                    ChunkCommand::Suspend => {
+                        pause.interrupt();
+                    }
+                    ChunkCommand::Resume | ChunkCommand::Stop => {
+                        pause.free();
+                        let _res = child_sender.send(command);
+                    }
+                }
+            }
+            Some(res) = finished_recv.recv() => {
+                let _ = jh.await;
+                match res {
+                    (Ok(0), cycles) => {
+                        return Ok(ChunkState::Completed(cycles));
+                    }
+                    (Ok(exit_code), _) => {
+                        return Err(ScriptError::validation_failure(
+                            &script_group.script,
+                            exit_code
+                        ))},
+                    (Err(err), _) => {
+                        return Err(ScriptError::VMInternalError(err));
+                    }
+                }
+
+            }
+            else => { break Err(ScriptError::validation_failure(&script_group.script, 0)) }
+        }
+    }
+}
+
+async fn run_vms_child(
+    mut machines: Vec<ResumableMachine>,
+    mut child_recv: watch::Receiver<ChunkCommand>,
+    finished_send: mpsc::UnboundedSender<(Result<i8, ckb_vm::Error>, u64)>,
+    context: Arc<Mutex<MachineContext>>,
+) {
```

Comment:
How about add `Result` to return type for this function, since it has many `.unwrap()` here.